File: java/src/main/java/com/android/frame/mvp/CommonBaseActivity.java
package com.android.frame.mvp;

import androidx.viewbinding.ViewBinding;

/**
 * Created by xuzhb on 2020/1/8
 * Desc:不需要额外声明View和Presenter的Activity的父类
 */
public abstract class CommonBaseActivity<VB extends ViewBinding> extends BaseActivity<VB, IBaseView, BasePresenter<IBaseView>> implements IBaseView {

    @Override
    public BasePresenter<IBaseView> getPresenter() {
        return new BasePresenter<>();
    }
}

File: kotlin/src/main/java/com/android/frame/mvp/CommonBaseActivity.kt
package com.android.frame.mvp

import androidx.viewbinding.ViewBinding

/**
 * Created by xuzhb on 2019/12/29
 * Desc:不需要额外声明View和Presenter的Activity的父类
 */
abstract class CommonBaseActivity<VB : ViewBinding> : BaseActivity<VB, IBaseView, BasePresenter<IBaseView>>(), IBaseView {
    override fun getPresenter(): BasePresenter<IBaseView> = BasePresenter()
}